streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -141,16 +144,17 @@ public KStreamImpl(KStreamBuilder topology, String name, Set<String> sourceNodes
 
         topology.addProcessor(name, new KStreamMap<>(mapper), this.name);
 
-        return new KStreamImpl<>(topology, name, null);
+        return new KStreamImpl<>(topology, name, sourceNodes, true);
```

Comment:
Should be `KStreamImpl<>(topology, name, null, true)` I think, since after the map the original topics may not be presenting the current stream any more (i.e. just what we did in the old code)?